modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/MulePluginJsonDescriptorGenerator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.resources;
+
+import static org.mule.runtime.deployment.model.internal.plugin.descriptor.PropertiesClassloaderDescriptor.PLUGINPROPERTIES;
+import static org.mule.runtime.deployment.model.internal.plugin.loader.AbstractPluginDescriptorLoader.MULE_PLUGIN_JSON_FILENAME;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.extension.api.resources.GeneratedResource;
+import org.mule.runtime.extension.api.resources.spi.GeneratedResourceFactory;
+import org.mule.runtime.module.extension.internal.model.property.ImplementingTypeModelProperty;
+
+import com.google.gson.Gson;
+
+import java.util.Optional;
+
+/**
+ * A {@link GeneratedResourceFactory} which generates a descriptor file and stores it in {@code JSON} format
+ *
+ * @since 4.0
+ */
+public final class MulePluginJsonDescriptorGenerator implements GeneratedResourceFactory {
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Optional<GeneratedResource> generateResource(ExtensionModel extensionModel) {
```

Comment:
Part of this class is duplicated on ArtifactPluginFileBuilder#createMulePluginJsonResource	and another test class. Clean it